x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/alerts_context.tsx

```diff
@@ -24,22 +25,40 @@ import type { AlertsTableImperativeApi } from '@kbn/response-ops-alerts-table/ty
  */
 const AlertsContext = createContext<{
   alertsTableRef: RefObject<AlertsTableImperativeApi>;
+  /**
+   * anonymization switch state in local storage
+   */
+  showAnonymizedValues: boolean;
+  setShowAnonymizedValues: React.Dispatch<React.SetStateAction<boolean | undefined>>;
 } | null>(null);
 
 const AlertsContextProviderComponent = ({ children }: PropsWithChildren) => {
   const alertsTableRef = useRef<AlertsTableImperativeApi>(null);
-  const contextValue = useMemo(() => ({ alertsTableRef }), []);
-  return <AlertsContext.Provider value={contextValue}>{children}</AlertsContext.Provider>;
+  const spaceId = useSpaceId();
+  const [showAnonymizedValues = false, setShowAnonymizedValues] = useLocalStorage<boolean>(
+    `securitySolution.aiAlertFlyout.showAnonymization.${spaceId ?? 'default'}`
```

Comment:
I would prefer not to fallback the spaceId in the local storage key. We had an [issue](https://github.com/elastic/kibana/pull/214794#discussion_r1998955291) that the spaceId was `undefined` and ended up target value not found. Could we render only when spaceId is available?

Reply:
Let me know how this looks: https://github.com/elastic/kibana/pull/218018/commits/13d3a1ad7cb35089d6986dcc0a99ae909830db53